src/main/java/net/neoforged/neoforge/items/IItemHandler.java

```diff
@@ -8,13 +8,29 @@
 import net.minecraft.world.Container;
 import net.minecraft.world.item.ItemStack;
 import net.neoforged.neoforge.fluids.FluidStack;
-import net.neoforged.neoforge.fluids.capability.IFluidHandler;
+import net.neoforged.neoforge.transfer.ResourceHandlerDeprecationHandling;
+import net.neoforged.neoforge.transfer.handlers.resources.IResourceHandler;
+import net.neoforged.neoforge.transfer.resources.IResource;
+import net.neoforged.neoforge.transfer.resources.ItemResource;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
 
+/**
+ * Implement this interface on your Block Entity if it can be used in a Hopper.
+ * This is used to interact with Hoppers, Droppers, Vanilla automation, etc.
+ *
+ * @deprecated Use {@link IResourceHandler} bound by type {@link ItemResource} instead
+ * 
+ *             <pre>{@code
+ *  IResourceHandler<ItemResource>
+ * }</pre>
+ */
+@Deprecated(since = ResourceHandlerDeprecationHandling.MC_1_21_6, forRemoval = true)
 public interface IItemHandler {
```

Comment:
There should be an implementation of `IItemHandler` that wraps an `IResourceHandler<ItemResource>` to help with migration at the use sites of the capability.